src/build_queue.rs

```diff
@@ -204,10 +205,10 @@ impl BuildQueue {
                     );
                     match res {
                         Ok(_) => debug!("{}-{} yanked", krate.name, krate.version),
-                        Err(err) => error!(
-                            "error while setting {}-{} to yanked: {}",
-                            krate.name, krate.version, err
-                        ),
+                        Err(err) => report_error(&anyhow!(err).context(format!(
```

Comment:
You can move the context up to where the operation happens to avoid the awkwardness here:

```rust
let res = conn.execute(...).with_context(|| {
    format!(
        "error while setting {}-{} to yanked",
        krate.name, krate.version
    )
});
```

Reply:
I'll give it a try later today, from what I remember in many places the code looked awkward after the change.

Reply:
Another option I've been meaning to try out in an `anyhow`-heavy codebase is to use something like [`fn-error-context`](https://docs.rs/fn-error-context) to allow just annotating functions with their context. That may require creating some extra functions though.

Reply:
I switched to using `.context` and `.with_context` and it doesn't look as bad as I thought it would. 

Let's see what you think :)